### PR TITLE
CNDIT-1609: Reorder notification topic submission

### PR DESCRIPTION
**Pull request by sveselev**
## Description
This PR includes fixes for the order of topics submission in investigation pre-processing service.

- **ProcessInvestigationDataUtil.java**: make the method for processing and sending notification data public
- **InvestigationService.java**: call processNotifications method after investigation topic has been sent

## JIRA
[CNDIT-1609](https://cdc-nbs.atlassian.net/browse/CNDIT-1609)

### CODE REVIEW

File: investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java [hunk 1]
```diff
@@ -98,6 +98,7 @@ public String processInvestigation(String value) {
                     InvestigationTransformed investigationTransformed = processDataUtil.transformInvestigationData(investigationData.get());
                     buildReportingModelForTransformedData(reportingModel, investigationTransformed);
                     pushKeyValuePairToKafka(investigationKey, reportingModel, investigationTopicReporting);
+                    processDataUtil.processNotifications(investigationData.get().getInvestigationNotifications(), objectMapper);
```
Comment (alog-enquizit) [inline investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java:101]: 👍

**Review by alog-enquizit — APPROVED** [review]
lgtm 👍

File: investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java [hunk 2]
```diff
@@ -50,12 +50,11 @@ public InvestigationTransformed transformInvestigationData(Investigation investi
         transformObservationIds(investigation.getObservationNotificationIds(), investigationTransformed, objectMapper);
         transformInvestigationConfirmationMethod(investigation.getInvestigationConfirmationMethod(), objectMapper);
         processInvestigationPageCaseAnswer(investigation.getInvestigationCaseAnswer(), investigationTransformed, objectMapper);
-        transformNotifications(investigation.getInvestigationNotifications(), objectMapper);
 
         return investigationTransformed;
     }
 
-    private void transformNotifications(String investigationNotifications, ObjectMapper objectMapper) {
+    public void processNotifications(String investigationNotifications, ObjectMapper objectMapper) {
```
Comment (upasanapattnaik-eq) [inline investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:57]: Do we have any logging message for Notification processing?

**Review by upasanapattnaik-eq — APPROVED** [review]
LGTM! Thank you for the update 👍

File: investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java (hunk 2 above)
Comment (sveselev) [reply investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:57]: Yes, but only if the investigation payload is missing notification data or there was an error processing the notification json. This is consistent with the logging for all other processing in the investigation preprocessing. We can create a story to improve this if needed.

Comment (upasanapattnaik-eq) [reply investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java:57]: Sounds good! We might want to add that in as a future enhancement.

**Pull request merged**